src/rust/tests/write.rs

```diff
@@ -138,3 +139,45 @@ fn geozero_to_fgb() -> Result<()> {
     fgb.write(&mut fout)?;
     Ok(())
 }
+
+#[test]
+fn test_save_fgb_and_load() {
+    let tempf = tempfile().expect("test failed");
+
+    // Save
+    let linestrings: Vec<LineString<f64>> = vec![
+        geo_types::line_string![
+            (x: -21.95156, y: 64.1446),
+            (x: -21.951, y: 64.14479),
+            (x: -21.95044, y: 64.14527),
+            (x: -21.951445, y: 64.145508)
+        ],
+        geo_types::line_string![(x: 0.0, y: 0.0), (x: 1.0, y: 1.0),],
+    ];
+
+    let mut fgb = FgbWriter::create("test_write", GeometryType::LineString, |_fbb, header| {
+        header.features_count = linestrings.len() as u64;
+        header.index_node_size = 0;
+    })
+    .expect("test failed");
+    fgb.set_crs(4326, |_fbb, _crs| {});
+
+    for geom in linestrings.iter() {
+        let geom: geo_types::Geometry<f64> = geom.to_owned().into();
+        fgb.add_feature_geom(geom, |_feat| {}).expect("test failed");
+    }
+    let mut file = BufWriter::new(&tempf);
+    fgb.write(&mut file).expect("test failed");
+
+    // Load
+    let mut filein = BufReader::new(&tempf);
```

Comment:
Reading the just written tempfile seems not to work. It's working with a regular file and an additional `file.flush()`.

Reply:
Using `NamedTempFile` now and can call `reopen` after `flush` to continue using temp file for read. Test passes.